### PR TITLE
[pkg/collector] Reorder platform initialization on collector

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Reorder `initializePlatform` call so that it happens as soon as possible during the rtloader's initialization.

### Motivation

- #5001 introduced the `windows_use_pythonpath` option to fix an issue where rtloader would crash while loading an incorrect Python version. However, in order to completely fix the issue that PR intended to address (which happens on Windows 2012 R2) we need to call the `initializePlatform` function  before the `make2`/`make3` functions.

### Describe your test plan

- Enable crash handling and check that the Agent runs correctly on Linux/macOS.
- On a Windows 2012 R2 machine install Python 2 and set the `PYTHONPATH` environment variable to point to this Python version. On the Agent configuration file set `python_version: 3`.  Manually run the Agent and check that it runs correctly.
- Repeat the previous steps but setting `windows_use_pythonpath` to true. Manually run the Agent and check that it throws an error similar to:

```
INFO | (pkg/collector/python/init.go:303 in Initialize) | Initializing rtloader with python3 C:\Program Files\Datadog\Datadog Agent\embedded3

Python path configuration:
  PYTHONHOME = 'C:\Program Files\Datadog\Datadog Agent\embedded3'
  PYTHONPATH = 'C:\Python27\Lib;C:\Python27\DLLs;'
  program name = 'python'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = 'C:\\Program Files\\Datadog\\Datadog Agent\\bin\\agent.exe'
  sys.base_prefix = 'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3'
  sys.base_exec_prefix = 'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3'
  sys.executable = 'C:\\Program Files\\Datadog\\Datadog Agent\\bin\\agent.exe'
  sys.prefix = 'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3'
  sys.exec_prefix = 'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3'
  sys.path = [
    'C:\\Python27\\Lib',
    'C:\\Python27\\DLLs',
    'C:\\Python27\\Lib\\lib-tk',
    'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3\\python38.zip',
    'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3\\DLLs',
    'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3\\lib',
    'C:\\Program Files\\Datadog\\Datadog Agent\\bin',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
  File "C:\Python27\Lib\encodings\__init__.py", line 123
    raise CodecRegistryError,\
                            ^
SyntaxError: invalid syntax

Current thread 0x0000089c (most recent call first):
<no Python frame>
```

### CODE REVIEW

**Review by albertvaka — COMMENTED** [review]
I think the Linux platform code might crash if it runs here, because it could call `C.handle_crashes(rtloader, C.int(cCoreDump))` when rtloader is not initialized yet. 

I think it could be refactored so `handleCrashes` is not a member of RTLoader (since it doesn't seem to use anything from it), but the easiest option is to move only the Windows code here while keeping the `initializePlatform` function where it was. Or to have two `initializePlatform` functions, one that runs before and one that runs after.

**Comment by mx-psi** [comment]
I am running Kitchen tests to check if the Agent would crash on Linux but yeah, it makes sense that it would crash, I didn't notice it was using `rtloader`. I am fine with either making the `HandleCrashes` method a standalone function or moving the code, let's see what the folks from Agent Core think.

**Comment by mx-psi** [comment]
I refactored the `handle_crashes` rtloader function so that it no longer needs to have an `rtloader` object for it to function.

File: rtloader/rtloader/api.cpp [hunk 1]
```diff
@@ -323,14 +337,82 @@ void clear_error(rtloader_t *rtloader)
...
+    int err = sigaction(SIGSEGV, &sa, NULL);
+
+    if (enable && err == 0) {
+        __sync_synchronize();
+        core_dump = core;
+    }
+    if (err) {
+        std::ostringstream err_msg;
+        err_msg << "unable to set crash handler: " << strerror(errno);
+        *error = strdupe(err_msg.str().c_str());
```
Comment (albertvaka) [inline rtloader/rtloader/api.cpp:412]: 👌

File: pkg/collector/python/init_nix.go [hunk 2]
```diff
@@ -31,8 +34,13 @@ func initializePlatform() error {
...
 		}
 
-		if C.handle_crashes(rtloader, C.int(cCoreDump)) == 0 {
-			log.Errorf("Unable to install crash handler, C-land stacktraces and dumps will be unavailable")
+		var handlerErr *C.char = nil
+		if C.handle_crashes(C.int(cCoreDump), &handlerErr) == 0 {
+			log.Errorf("Unable to install crash handler, C-land stacktraces and dumps will be unavailable: %s", C.GoString(handlerErr))
+			if handlerErr != nil {
+				// pyErr tracked when created in rtloader
+				C._free(unsafe.Pointer(handlerErr))
```
Comment (albertvaka) [inline pkg/collector/python/init_nix.go:42]: 👌

**Review by albertvaka — APPROVED** [review]
👍 💯

**Review by olivielpeau — APPROVED** [review]
Just a doc nit, but other than that looks great! Feel free to merge once addressed

**Pull request merged**